### DISCUSSION TITLE
Clog with Ceramic

### DISCUSSION THREAD

**Discussion by bpecsek** [rabbibotton/clog#54] [General]
I've been trying to make ceramic work with clog but can't figure out how.

The ceramic window pops up but nothing is shown.

Has anyone made an example that works with ceramic?

**Comment by bpecsek** [comment]
I think I kind of figured it out but not really.

It is a start, but unfortunately, when I close the electron window the electron app still running in the background and the usocket port stays open.

The other problem is that when I specify :static-root in initialize, nothing shows up in the electron windows.

It would be nice to see a properly working example from someone.

ceramic-hello-clog.asd
```
(defsystem ceramic-hello-clog
  :author "Bela Pecsek <bela.pecsek@gmail.com>"
  :maintainer "Bela Pecsek <bela.pecsek@gmail.com>"
  :license "MIT"
  :version "0.0.1"
  :depends-on (:ceramic
	       :clog)
  :components ((:file "ceramic-hello-clog"))
  :description "A Hello World example for Ceramic with Clog.")
```
ceramic-hello-clog.lisp
```
(defpackage #:ceramic-hello-clog
  (:use #:cl #:clog)
  (:export main))
(in-package :ceramic-hello-clog)

(defun on-new-window (body)
  (create-div body :content "Hello Clog!")
  (run body))

(defun start-app ()
  (initialize #'on-new-window))

(defun main ()
  (ceramic:start)
  (let ((window (ceramic:make-window :title "A simple App"
				     :width 800
				     :height 600
				     :url (format nil "http://localhost:~D/" 8080))))
    (ceramic:show window)
    (start-app)))

(ceramic:define-entry-point :ceramic-hello-world ()
  (main))
```

**Reply by rabbibotton** [reply]
I hope you will consider sending me a small sample app with some comments to add as a tutorial or demo :)

**Reply by rabbibotton** [reply]
Take a look at NATIVE.md with instructions for ceramic.

**Reply by rabbibotton** [reply]
btw, NATIVE.md is though missing a (clog:shutdown) and when get a chance I'll see how to hook the close of electron. What I really need to do is create though a gtk example, a win64 example and a macox example, is easy enough to do just time...

**Reply by rabbibotton** [reply]
I updated NATIVE.md with the needed code to shutdown everything properly.

5. We need to add to the botton of on-new-window code to shutdown app.

```
  (clog:run body) ; wait while body is running
  (ceramic:quit)  ; quit ceramic/electron
  (clog:shutdown) ; shutdown clog
```

**Reply by rabbibotton** [reply]
Added full project at - https://github.com/rabbibotton/elect

**Comment by mmontone** [comment]
Hi.

Just wanted to report that I'm successfully using this as an alternative: https://github.com/webview/webview
It is very lightweight and straightforward. Only tried on Linux, though.

**Reply by rabbibotton** [reply]
I will take a look and see about adapting it to what is in NATIVE.md of course if you want to give a try... would be cool :)
